great_expectations/data_context/types/base.py

```diff
@@ -976,6 +891,108 @@ def make_datasource_config(self, data, **kwargs):
         return DatasourceConfig(**data)
 
 
+class DatasourceConfig(BaseYamlConfig):
```

Comment:
Given this type does not inherit from `BaseYamlConfig` today, can we avoid introducing it? Eventually we don't want any of our config objects to inherit from BaseYamlConfig